### ISSUE TITLE
Add Morpheus workflow example

### ISSUE THREAD

**Issue by jacobtomlinson** [rapidsai/deployment#476]
Add an example notebook demonstrating a Morpheus workflow.

Outline:

- Data source
- Building a morpheus application
- Morpheus containers
- Deploying the workflow

Questions to answer before building the workflow:

- What are common data sources? Kafka streams?
- What is a good Morpheus use case to demonstrate?
- What platforms are Morpheus users commonly deploying on? Kubernetes on Cloud? Something else?

**Comment by ncclementi** [comment]
Taking some notes after looking into this a bit. In an exploratory mode trying to understand what kind of docs we would like in the RAPIDS deployment doc.  We want to understand this questions first: 

- What are common data sources? Kafka streams?
- What is a good Morpheus use case to demonstrate? 
    - Is there something simple to showcase kafka + Morpheus + RAPIDS, and is this a good example?
    - maybe something with real-time monitoring and threat detection of network traffic?
- What platforms are Morpheus users commonly deploying on? Kubernetes on Cloud? Something else? 
     - What are the requirements?
     - Are there any docs on how to deploy? 

I've been looking at: https://github.com/nv-morpheus/Morpheus/blob/branch-25.06/docs/source/examples.md and there are some interesting things, but it's a bit hard from each case to understand how things work, or put all the pieces together. 

Notes and extra questions:
- Would the best approach be starting from a rapids+jupyter notebooks image and installing morpheus there via conda? (is morpheus-core enough)
- There are also [pre-built morpheus images](https://github.com/nv-morpheus/Morpheus/blob/5fe1e7ded81aa541d66cda5cd431ffaf7c47c129/docs/source/getting_started.md#pull-the-morpheus-image) but it's unclear starting from there and then adding rapids and jupyter is the way to go. 
- The examples form the link above require a whole set of extra dependencies, that seem to be the kind of environment that is one-fits-all see [recommended environment]( https://github.com/nv-morpheus/Morpheus/blob/5fe1e7ded81aa541d66cda5cd431ffaf7c47c129/conda/environments/examples_cuda-128_arch-x86_64.yaml)
- There are some interesting things in the [morpheus experimental repo](https://github.com/nv-morpheus/morpheus-experimental/tree/5c71ca54ad46cd9b9916f97181ca6438e0755625) but I can pin point how is morpheus being used in each example, or if this is a good place to look.

**Comment by mdemoret-nv** [comment]
>What are common data sources? Kafka streams?

Kafka is generally whats used in production, but its easier to get started just reading from a file

>What is a good Morpheus use case to demonstrate?

Our Sensitive Information Detection (SID) example is a good one to get started with. It can be run with Kafka + Morpheus + RAPIDS (Thats true of every example since cuDF is built into our base message classes.)

> What platforms are Morpheus users commonly deploying on? Kubernetes on Cloud? Something else?

It can vary. Would need to ask some SAs what the common platform is.

> Would the best approach be starting from a rapids+jupyter notebooks image and installing morpheus there via conda? (is morpheus-core enough)

Yes, I think this would be fine.

>The examples form the link above require a whole set of extra dependencies, that seem to be the kind of environment that is one-fits-all see [recommended environment](https://github.com/nv-morpheus/Morpheus/blob/5fe1e7ded81aa541d66cda5cd431ffaf7c47c129/conda/environments/examples_cuda-128_arch-x86_64.yaml)

Yes, that environment will allow you to run any examples. However, for SID, I do not believe you need any extra dependencies outside of morpheus-core. (Need to double check that)

>There are some interesting things in the [morpheus experimental repo](https://github.com/nv-morpheus/morpheus-experimental/tree/5c71ca54ad46cd9b9916f97181ca6438e0755625) but I can pin point how is morpheus being used in each example, or if this is a good place to look.

I wouldnt recommend using these since they are generally maintained less and less documented.

**Comment by ncclementi** [comment]
Thank you @mdemoret-nv, this is very helpful, few more questions:

> Kafka is generally whats used in production, but its easier to get started just reading from a file

This might be a good starting point then, we can start with a file and then move to kafka as a followup example. Unless we want to show the full process, @jacobtomlinson do you have any particular direction in mind? 

> Our Sensitive Information Detection (SID) example is a good one to get started with. It can be run with Kafka + Morpheus + RAPIDS (Thats true of every example since cuDF is built into our base message classes.)

I'm assuming the SID example you reference is this https://github.com/nv-morpheus/Morpheus/tree/branch-25.06/examples/nlp_si_detection  

I don't think we can make the connection right away for users that this is a rapids + morpheus example, I understand that `cudf` might be used under the hood (though I'm not able to find where ), but when I read the example, it seems to be like a black box example ready to go and pure cli driven. Maybe we need to align more on what kind of example we want to show , but in my head I'd like to be able to get an example that can serve as a model for a user to then adapt and do their own thing.  

The example above is a bit of a black box to me (unless I'm missing something) that if I want to adapt, or change a bit I can't figure out how. 

I think we need to answer: Who is the target user for this documentation  page?

**Comment by jacobtomlinson** [comment]
> do you have a particular direction in mind?

I'm keen for us to document whatever makes sense. We can definitely start with a file, but given this is for the deployment documentation I'm keen to include a Kafka deployment in there too. But equally don't be bound by the image in my head if evidence shows we should do something else.

> Who is the target user for this documentation page>

The target user is folks using RAPIDS libraries in the context of Morpheus. I had seen examples like [this](https://github.com/nv-morpheus/Morpheus/blob/branch-25.06/examples/gnn_fraud_detection_pipeline/stages/graph_sage_stage.py#L62-L86) where `cudf` is being used within a stage. 

I'm keen for our list of workflow examples to include more than just XGBoost HPO examples. Adding some stream processing examples using Morpheus would be a nice addition.

That being said our documentation should be a jumping off point into the Morpheus documentation, but we should have enough in ours to show the art-of-the-possible for deploying a streaming pipeline that uses the Python stages.

**Comment by ncclementi** [comment]
> but given this is for the deployment documentation I'm keen to include a Kafka deployment in there too.

I agree, I'm trying to find some info on how to set this up. But I imagine we want to have kafka and a rapids + morpheus deployment probably in K8s. 

> The target user is folks using RAPIDS libraries in the context of Morpheus. I had seen examples like [this](https://github.com/nv-morpheus/Morpheus/blob/branch-25.06/examples/gnn_fraud_detection_pipeline/stages/graph_sage_stage.py#L62-L86) where cudf is being used within a stage.

Agreed

**Comment by ncclementi** [comment]
More info on kafka, taking notes here from what I'm learning. 

- We need to setup a kafka server,  Unclear if we should do this as part of a k8s cluster or we can probably do all in one docker container. 
    - Locally (or on VM?) via docker:  https://kafka.apache.org/quickstart. 
        - We would probably add rapids and morpheus to the image.
        - We would start from this image https://hub.docker.com/r/apache/kafka
   - Some form of k8s deployment
        - K8s  + kafka blog deployment: https://phoenixnap.com/kb/kafka-on-kubernetes 
        - there is some gke + kafka docs (I think this is more than what we need): https://cloud.google.com/kubernetes-engine/docs/tutorials/stateful-workloads/kafka
- We need producer get data into kafka topic: There is kafka cli,  but I think we should do this via python-kafka, 
    - there is a nice function in morpheus but it's private but it gives us an idea on how to create a producer from a file: https://github.com/nv-morpheus/Morpheus/blob/5fe1e7ded81aa541d66cda5cd431ffaf7c47c129/tests/_utils/kafka.py#L130-L169
   - Also this test could help as inspiration. https://github.com/nv-morpheus/Morpheus/blob/5fe1e7ded81aa541d66cda5cd431ffaf7c47c129/tests/morpheus/apps/test_phishing_kafka.py#L34
    - Need to find some data that makes sense to use with streaming.
- as part of our pipline we will have a KafkaSourceStage from morpheus, this acts as a consumer 
- have other stages to do something with the data, using rapids

**Comment by ncclementi** [comment]
Closed via https://github.com/rapidsai/deployment/pull/555
